packages/babel-core/src/config/config-chain.js

```diff
@@ -712,3 +744,33 @@ function matchPattern(
   }
   return pattern.test(pathToTest);
 }
+
+function printChain(
+  chain: ConfigChain,
+  formatter: $Values<typeof ChainFormatter>,
+  context: ConfigContext,
+  validatedFile?: ValidatedFile,
+): string {
+  function formatHeading(type, context, validatedFile) {
+    let heading = "";
+    if (type === ChainFormatter.Programmatic) {
+      heading = "programmatic options";
+      if (context.caller?.name) {
+        heading += " from " + context.caller.name;
```

Comment:
It will print `programmatic options (from babel-loader)` when caller name is given.